File: chapitre7/exercice7.1.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <signal.h>

int main()
{
    pid_t pid = fork();

    if (pid == 0)
    {
        while (1)
        {
            printf("Fils en cours d'exécution...\n");
            sleep(1);
        }
    }
    else
    {
        char choix;

        while (1)
        {
            printf("\nMenu:\n");
            printf("s: stop\n");
            printf("r: resume\n");
            printf("q: quit\n");
            printf("Choix: ");

            scanf(" %c", &choix);

            if (choix == 's')
            {
                kill(pid, SIGSTOP);
            }
            else if (choix == 'r')
            {
                kill(pid, SIGCONT);
            }
            else if (choix == 'q')
            {
                kill(pid, SIGKILL);
                break;
            }
        }
    }

    return 0;
}

File: chapitre7/exercice7.1.py
import os
import signal
import time

pid = os.fork()

if pid == 0:
    # FILS
    while True:
        print("Fils en cours d'exécution...")
        time.sleep(1)

else:
    # PERE
    while True:
        print("\nMenu:")
        print("s: stop")
        print("r: resume")
        print("q: quit")

        choix = input("Choix: ")

        if choix == 's':
            os.kill(pid, signal.SIGSTOP)

        elif choix == 'r':
            os.kill(pid, signal.SIGCONT)

        elif choix == 'q':
            os.kill(pid, signal.SIGKILL)
            break